src/main/io/serial_4way.c

```diff
@@ -464,36 +463,36 @@ void esc4wayProcess(serialPort_t *mspPort)
         // restart looking for new sequence from host
         do {
             CRC_in.word = 0;
-            timedOut = ReadByteCrc(&ESC, ESC_TIMEOUT_US);
-        } while ((ESC != cmd_Local_Escape) && !timedOut);
+            // No timeout as BLHeliSuite32 has this loops sitting indefinitely waiting for input
+            ReadByteCrc(&ESC, 0);
+        } while (ESC != cmd_Local_Escape);
 
         RX_LED_ON;
 
         Dummy.word = 0;
         O_PARAM = &Dummy.bytes[0];
         O_PARAM_LEN = 1;
 
+        timedOut = ReadByteCrc(&CMD, CMD_TIMEOUT_US);
+        timedOut |= ReadByteCrc(&ioMem.D_FLASH_ADDR_H, ARG_TIMEOUT_US);
```

Comment:
`timedOut  = timedOut  || ReadByteCrc(&ioMem.D_FLASH_ADDR_H, ARG_TIMEOUT_US);` will short-circuit on first timeout. And possibly make code a bit simpler

Reply:
Thanks. I've implemented a change inspired by this.